a10_octavia/db/repositories.py

```diff
@@ -303,12 +303,17 @@ def get_all_deleted_expiring(self, session, exp_age):
         return id_list
 
     def get_partition_for_project(self, session, project_id):
-        return session.query(self.model_class).filter(
-            self.model_class.project_id == project_id).first().partition_name
+        vthunder_project = self.get_vthunder_by_project_id(session, project_id)
+        if vthunder_project:
```

Comment:
This should never be an issue. We should never store a Thunder entry unless it has a partition name already

Reply:
This was the issue occurring due to the revert statement which I removed "CreateRackVthunderEntry". Since I removed that this condition is not required here.
will remove it.